### ISSUE TITLE
oauth2 not display 

### ISSUE THREAD

**Issue by sabarsaran** [mouneyrac/moodle-auth_googleoauth2#103]
Hi.

i have configured the plugin with oauth2 details still it is not displaying the login buttons in login page.

can you please help

thanks

**Comment by sabarsaran** [comment]
Can  you plz help me guys.....

**Comment by PickledKitty** [comment]
If you View Page Source, does the code for the buttons show up? ( it comes up under <center></center> )

**Comment by cryptosmith** [comment]
I may have a variant of this. I'm using the Essential theme and I've activated Facebook for oauth2 login. When I select "login" from a page, it takes me to the login page. A Facebook button flashes briefly at the top of the page. Then the page is redrawn with the default oauth-free login information.

**Comment by cryptosmith** [comment]
More information - While doing other things, not logged in, Moodle put up that simple page it has to signal a slow operation, saying something about "... in progress" with a spinning icon. At the top of the page appeared the missing Facebook login logo. It disappeared, replaced by the regular login page, once the slow operation completed.

**Comment by enlp** [comment]
Hi i have configured it correctly. but my problem is that the login icons showup at the very top of the page which is obcured by theme "essential" 's header strip. My question is how to make the buttons visible at the bottom of the page just below the login box?????? thanks in advance

**Comment by mouneyrac** [comment]
Yes its normal the feature to display the icons "works" on Clean theme because it's kind of a trick as soon as you change the theme its likely to be ugly. One solution is to insert the code line yourself in your theme as explained in the settings page. Otherwise, for the coders, one the solution could be to insert the code by JS if it is possible to load JS at this point. I'm unsure of that.

**Comment by cryptosmith** [comment]
How is the "Clean" theme different from other themes, such that this module works there, but nowhere else?

I did not see documentation for this plugin to suggest that it only works with one Moodle theme.

How do other plugins adapt to different themes?

**Comment by mouneyrac** [comment]
Hi Cryptosmith,

>>> How is the "Clean" theme different from other themes, such that this module works there, but nowhere else?
the feature is a bit of a hack. If I remember well I tried this hack when I first started to write the plugin but it was a bit hacky so I didn't implement it. But it was contributed later so I integrated it. How does the hack work? Basically it displays the buttons before the rest of the page as at the moment the code is executed before the rest of the page is  renderer. The funny part is that Clean (and all themes doing the same thing a Clean) does force the header bar to be on the top, so the buttons appear to be displayed correctly on Clean theme and similar theme.

>>> I did not see documentation for this plugin to suggest that it only works with one Moodle theme.
I totally agree. I'll mention it in a future update :)

>>> How do other plugins adapt to different themes?
There are a couple of Oauth2 plugin on Moodle.org plugin repitory. I have just discovered a new one and this one does use html injection by js as I mentioned in my previous comment before I knew about the plugin. It's likely going to work with your custom theme. I would encourage you to try all different alternatives to see what suit you best. 
They may be even better solution than the html injection, need to check other auth plugins :)

Cheers,
Jerome
